tests/core/account_builder_tests.cpp

```diff
@@ -0,0 +1,314 @@
+/* 
+File: account_builder_tests.cpp
+
+Brief:  Account builder unit tests
+
+Authors: Daniel Mendez 
+         Alex Salazar
+         Arman Alauizadeh 
+         Alexander DuPree
+         Kyle Zalewski
+         Dominique Moore
+
+https://github.com/AlexanderJDupree/ChocAn
+ 
+*/
+
+#include <cstring>
+#include <stack>
+#include <catch.hpp>
+#include <ChocAn/data/mock_db.hpp>
+#include <ChocAn/core/utils/account_builder.hpp>
+
+const std::string valid_name("firstName,lastName");
+const std::string valid_address("123 Street,Portland,OR,97080");
+const std::string valid_type("member");
+const std::string no_msg("");
+Build_Instructions building_phase;
+Data_Gateway::Database_Ptr db = std::make_unique<Mock_DB>();
+
+TEST_CASE("Account builds with valid inputs", "[account_builder]")
```

Comment:
These tests need to broken up into different sections. Purists will say that there should be AT MOST one assertion per section, however i'm flexible with that. 15+ assertions in a section though is a little much. A lot of these assertions can be grouped together into one logical statement with some refactoring. For example, `get_current_issues` can just return an optional type so you don't need to check if the error_msg is no_msg AND the info vector is empty. You can just check if the optional type is populated or not. 

Also you can nest test sections, so if you're building up input to build an account object you can nest those sections and retain state.

Reply:
the new tests will be implemented differently. i wont be checking for error message, just that no exception or the right exception is thrown. i did nest the sections in the Invalid User input test case, since it captures an actual use case process. i plan to keep the nested sections in the refactor, although the content will change to reflect the new logic.

Reply:
i pushed it to two in some sections because you said you were flexible and it represented the logic of before/after style testing pretty well. for the most part i limited the sections to one assert.